DirectProgramming/DPC++FPGA/ReferenceDesigns/decompress/src/snappy/snappy_reader.hpp

```diff
@@ -150,8 +150,8 @@ unsigned SnappyReader(unsigned in_count) {
   unsigned literal_len_counter;
 
   unsigned data_read = data_read_in_preamble;
-  bool all_data_read = data_read < in_count;
-  bool all_data_read_next = data_read < in_count - literals_per_cycle;
+  bool all_data_read = data_read >= in_count;
+  bool all_data_read_next = data_read >= in_count - literals_per_cycle;

```

Comment:
This change (from < to >=) seems pretty fundamental - how did the code work before this change AND after it?!?

Reply:
I'm not entirely sure how it was working before 
This `all_data_read` is not stopping the loop, it is just used in the case where you don't read anything from the pipe (non blocking read) so that we may do something else.
These two values are initialisation values that are overwritten by the next two pipe reads in the computation loop.
I'm guessing that the first two iterations always ended up having something to read from the pipe which made these false values being overwritten without ever being used.